cgrok/grok_match.c

```diff
@@ -52,8 +52,7 @@ int grok_match_walk_next(const grok_match_t *gm,
   }
 
   *namelen = gct->name_len;
-  *name = malloc(*namelen);
-  memcpy(*name, gct->name, *namelen);
+  *name = gct->name;
```

Comment:
Unfortunately the change where this was introduced is a big mess: https://github.com/jordansissel/grok/commit/c4a402755ed79e322376ff11dd616e2b5187f4be

I will add a strongly worded comment.